nuc_morph_analysis/lib/preprocessing/labeling_neighbors_helper.py

```diff
@@ -0,0 +1,386 @@
+import numpy as np
+import pandas as pd
+from scipy.ndimage import binary_dilation
+
+"""
+Example for how to find neighbors of a breakdown event
+
+1. corrects cellIds with missing neighbors to avoid errors
+2. it looks for all breakdown events (predicted_breakdown), and marks them in the columns frame_of_breakdown as True.
+3. then select rows (i.e. CellIds) where frame_of_breakdown==True, identify all the neighbors for those cells
+4. create a new columns called has_mitotic_neighbor_breakdown and mark it as True for all of those neighbors.
+5. Then, because the mitotic cell will lack a segmentation after breakdown, create a new column called has_mitotic_neighbor_forward_dilated. 
+For each track_id, expand all has_mitotic_neighbor_breakdown==True values forward in time by 45 minutes 
+(which is the upper bound for mitotic durations in these colonies)
+(this is repeated for frame_of_formation but to expand the True values backwards in time)
+"""
+
+def correct_cellids_with_missing_neighbors(df):
+    """
+    Function to correct the `neighbors` column in the dataframe
+    where the `neighbors` column is a string representation of a list
+    but some of the entries are 'None' instead of '[]'
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        the dataframe with the `neighbors` column
+
+    Returns
+    -------
+    df : pd.DataFrame
+        the dataframe with the corrected `neighbors` column
+    """
+
+    # one CellId is weird and has neighbors = None, so we need to change that
+    # df.loc['ed8124da9dfe45bc3b64d65aeb7446ff70db4255d41309bb5d1eb9b4','neighbors'] = '[]'
```

Comment:
Should this be deleted? Or is the comment still correct?

Reply:
It was one CellId in the baseline dataset (i.e. "small","medium","large" colonies), but the "feeding_control" dataset has more CellIds that have no df['neighbors']=None

Reply:
Correction. There was only one mitotic CellId that had neighbors == None. there are multiple in the baseline colony datasets as well.